### PR TITLE
add a simple go application to demo acceleration

### PR DESCRIPTION
**Pull request by y-f-u**
Add a sample app to demonstrate the use case of spiced accelerating queries

### CODE REVIEW

File: acceleration/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# Sample
+
+This sample showcases the way to use spiced to simplify the process of building a CQRS pattern for your application.
```
Comment (phillipleblanc) [inline acceleration/README.md:3]: What is CQRS?

Comment (phillipleblanc) [reply acceleration/README.md:3]: Explain acronyms before using them

Comment (phillipleblanc) [inline acceleration/README.md:3]: Don't call it `spiced`, let's use the same wording we use in the Docs/Readme

File: acceleration/main.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+	"github.com/spiceai/sample-1/pkg/flightsql"
+
+	_ "github.com/lib/pq"
+)
+
+var PRODUCTS = []string{"the most beautiful flower", "the most delicious chocalate", "nintendo"}
+
+func main() {
+	fmt.Println("Start the exmaple application to use Spice")
```
Comment (phillipleblanc) [inline acceleration/main.go:19]:
```suggestion
	fmt.Println("Start the example application to use Spice")
```

File: acceleration/README.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
+# Sample
```
Comment (phillipleblanc) [inline acceleration/README.md:1]: It was not obvious to me in my first pass that this sample chains Spice instances together - that needs an explanation and it would be good to have a figma to illustrate the architecture here, to make it easier to understand.

File: acceleration/README.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+- /rankings-materialized - fetch data from spiced runtime with materialized ranking view, it pulls data from ranking view from the previous spiced runtime and accelerates it locally
+
+## How to run
+
+`docker-compose up --build`
+
+then call the rankings API
+`curl localhost:9999/rankings`
+
+## Performance benchmark using oha
```
Comment (phillipleblanc) [inline acceleration/README.md:21]: It would be good to have a summary of the results in a table format - as it is, its hard to parse the details quickly.

File: acceleration/Dockerfile.spiceai [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN git clone https://github.com/spiceai/spiceai.git
+
+RUN cd spiceai && make -C bin/spiced
+
+FROM debian
+
+RUN apt-get update && apt-get install -yqq libssl-dev sqlite3
+
+COPY --from=build /spiceai/target/release/spiced /usr/local/bin
```
Comment (phillipleblanc) [inline acceleration/Dockerfile.spiceai:13]: Could we have this pull the latest release from GitHub instead of compiling? That would cut down on the time needed to build this. (We could also update in the next release to pull from the official docker image)

File: acceleration/README.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
+# Sample
+
+This sample showcases the way to use spiced to simplify the process of building a CQRS pattern for your application.
+
+In this sample, we have a `/orders` API which creates a random order with a random count.
```
Comment (phillipleblanc) [inline acceleration/README.md:5]: Does this sample need to call the `/orders` API for it to work? If so, we should show that step here (or even better, just pre-populate on init)

File: acceleration/spicepod-materialized.yaml
Comment (phillipleblanc) [inline acceleration/spicepod-materialized.yaml:1]: This seems like a workaround to have a materialized view? I would actually just remove this completely - it simplifies the sample conceptually. Ideally we support this directly by letting people set the `acceleration` property on a view.

Comment (phillipleblanc) [reply acceleration/spicepod-materialized.yaml:1]: But if you feel strongly about leaving this in, then I would make it clear what this is doing - i.e. working around the fact that we don't have materialized views yet.

File: acceleration/docker-compose.yaml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    expose:
+      - "50051"
+    deploy:
+      resources:
+        limits:
+          cpus: '2'
+          memory: 2G
+        reservations:
+          cpus: '2'
+          memory: 2G
```
Comment (phillipleblanc) [inline acceleration/docker-compose.yaml:28]:
```suggestion
```

Unless its required to make the sample work, let's remove this to simplify it

File: acceleration/docker-compose.yaml
Comment (phillipleblanc) [inline acceleration/docker-compose.yaml:1]: Should we rename this to `compose.yaml`?

https://docs.docker.com/compose/compose-application-model/#the-compose-file

File: acceleration/Dockerfile.spiceai (hunk 5 above)
Comment (phillipleblanc) [reply acceleration/Dockerfile.spiceai:13]: Actually for this to work properly on M1 macs, we need to build a Linux aarch64 version of spiced

File: acceleration/docker-compose.yaml (hunk 7 above)
Comment (y-f-u) [reply acceleration/docker-compose.yaml:28]: this is to make the benchmark more legit

File: acceleration/spicepod-materialized.yaml
Comment (y-f-u) [reply acceleration/spicepod-materialized.yaml:1]: yes, normally people would build a materialised view which is reflected in the benchmark in readme. It's about 5 times faster for retrieving from there. Will add more explanation.

File: acceleration/README.md (hunk 1 above)
Comment (digadeesh) [reply acceleration/README.md:3]: Please resolve / address in next iteration.  Closing comment to merge.

Comment (digadeesh) [reply acceleration/README.md:3]: Please resolve / address in next iteration.  Closing comment to merge.

File: acceleration/main.go (hunk 2 above)
Comment (digadeesh) [reply acceleration/main.go:19]: Please resolve / address in next iteration.  Closing comment to merge.

File: acceleration/README.md (hunk 4 above)
Comment (digadeesh) [reply acceleration/README.md:21]: Please resolve / address in next iteration.  Closing comment to merge.

File: acceleration/Dockerfile.spiceai (hunk 5 above)
Comment (digadeesh) [reply acceleration/Dockerfile.spiceai:13]: Please resolve / address in next iteration.  Closing comment to merge.

File: acceleration/README.md (hunk 6 above)
Comment (digadeesh) [reply acceleration/README.md:5]: Please resolve / address in next iteration.  Closing comment to merge.

File: acceleration/spicepod-materialized.yaml
Comment (digadeesh) [reply acceleration/spicepod-materialized.yaml:1]: Please resolve / address in next iteration.  Closing comment to merge.

File: acceleration/docker-compose.yaml (hunk 7 above)
Comment (digadeesh) [reply acceleration/docker-compose.yaml:28]: Please resolve / address in next iteration.  Closing comment to merge.

**Comment by digadeesh** [comment]
Resolving to merge - will create new item to address comments.

**Review by digadeesh — APPROVED** [review]

File: acceleration/README.md (hunk 3 above)
Comment (digadeesh) [reply acceleration/README.md:1]: Please resolve / address in next iteration.  Closing comment to merge.

File: acceleration/docker-compose.yaml
Comment (digadeesh) [reply acceleration/docker-compose.yaml:1]: Please resolve / address in next iteration.  Closing comment to merge.
